gensim/models/_fasttext_bin.py

```diff
@@ -160,6 +161,16 @@ def _load_vocab(fin, new_format, encoding='utf-8'):
     """
     vocab_size, nwords, nlabels = _struct_unpack(fin, '@3i')
 
+    #
+    # We must use backslashreplace instead of replace or ignore here, because
+    # we must avoid collisions in the decoded word, e.g.
+    # https://github.com/RaRe-Technologies/gensim/issues/2402
+    #
+    # Unfortunately, backslashreplace is only available on Py3.  On Py2, we
+    # can't really do anything to avoid collisions.
```

Comment:
Can we calculate the number of collisions, and adjust the `assert` accordingly?  What functionality hinges on the difference between "actual vectors after loading" vs "declared vectors in file before loading"?

Reply:
I've rewritten the code to avoid collisions.